Utilities/bootstrap

```diff
@@ -1175,6 +1186,10 @@ def main():
     # Append the versioning build flags.
     build_flags.extend(create_versoning_args(args))
 
+    # Don't use GNU strerror_r on Android.
+    if 'ANDROID_DATA' in os.environ:
+        build_flags.extend(["-Xswiftc", "-Xcc", "-Xswiftc", "-U_GNU_SOURCE"])
```

Comment:
This is fine for now but this should go to BuildPlan.swift so swiftpm can correct compile itself and other Swift targets for Android.

Reply:
Unfortunately, `_GNU_SOURCE` was set for all clang imports for Android in the Swift compiler a year ago, apple/swift@60e2205, so changing that for all packages built for Android could break stuff. This tweak simply unsets it so the GNU `strerror_r` isn't used when building SPM itself, and avoids those larger issues.